cli/routerkeygen-cli.cpp

```diff
@@ -9,15 +9,116 @@
 
 #include "qcmdlineparser/qcmdlineparser.h"
 
+static void generateCode(QString &ssid, QString &mac)
+{
+    QScanResult wifi(ssid, mac);
+    WirelessMatcher wm;
+
+    // Add keygens
+    wifi.checkSupport(wm);
+
+    // Calculate!
+    QVector<Keygen *> * keygens = wifi.getKeygens();
+    if (keygens == nullptr)
+        return; /* Out of memory */
+
+    // Print keys
+    // Format is MAC<TAB>Password
+    for (int i = 0; i < keygens->size(); ++i) {
+        try {
+            QVector<QString> r = keygens->at(i)->getResults();
+            foreach (QString s, r) {
+                std::cout<< mac.toUtf8().data() << '\t' << s.toLatin1().data() << std::endl;
+            }
+        } catch (int e) {
+            continue;
+        }
+    }
+}
+
+static QString hexDecode(const QString &in)
+{
+    auto a = QByteArray::fromHex(in.toUtf8().data());
+
+    return QString::fromStdString(a.toStdString());
+}
+
+static inline bool hexValid(const QString &in)
+{
+    if ((in.size() % 2) != 0) {
+        return false;
+    }
+
+     foreach (QChar c, in) {
+        if (!isxdigit(c.toLatin1())) {
+            return false;
+        }
+    }
+
+    return true;
+}
+
 int main(int argc, char * argv[]) {
     QCoreApplication app(argc, argv);
     QCmdLineParser parser;
     parser.addOption("-s", QCmdLineArgument::StoreValue, QObject::tr("SSID"), "--ssid", "network_name");
     parser.addOption("-m", QCmdLineArgument::StoreValue, QObject::tr("MAC address"), "--mac", "mac_address");
     parser.addOption("-k", QCmdLineArgument::StoreTrue, QObject::tr("Print keygen in form kgname:candidate"), "--kg");
     parser.addOption("-q", QCmdLineArgument::StoreTrue, QObject::tr("Print only calculated keys"), "--quiet");
+    parser.addOption("-r", QCmdLineArgument::StoreTrue, QObject::tr("Read SSID and MAC from stdin in hex"), "--read");
     QString error = "";
     QVariantMap options = parser.parse(QCoreApplication::arguments(), &error);
+
+    if (options.contains("r")) {
+        enum {
+            IDX_MAC  = 0,
+            IDX_SSID = 1,
+            IDX_MAX  = 2,
+        };
+        std::string line;
+
+        // Read line by line
+        // Format is: MAC<TAB>SSID
```

Comment:
This, along with output format should go in --help , since it's not obvious for the users.